### PR TITLE
Add new rules for no-debug, no-skip and no-only

### PR DESCRIPTION
**Pull request by benmonro**
https://github.com/miherlosev/eslint-plugin-testcafe/issues/8

### CODE REVIEW

**Comment by benmonro** [comment]
@miherlosev added the 3 new rules.  let me know when this can be merged and published as we could use it right away.

**Comment by miherlosev** [comment]
Hi @benmonro 

Thank you a lot for your contribution. I cloned your PR and run tests. There is one problematic test.
image

Please fix it.

**Comment by benmonro** [comment]
Can you please give me the full error messages you are getting? @miherlosev

**Comment by miherlosev** [comment]
C:\forks\for_review\master>npm run test

> eslint-plugin-testcafe@0.2.1 test C:\forks\for_review\master
> jest

(node:17080) ExperimentalWarning: The fs.promises API is experimental
(node:18104) ExperimentalWarning: The fs.promises API is experimental
(node:1568) ExperimentalWarning: The fs.promises API is experimental
(node:1164) ExperimentalWarning: The fs.promises API is experimental
 FAIL  tests/index.test.js
  ● should have all the rules

    expect(received).toMatchSnapshot()

    Snapshot name: `should have all the rules 1`

    - Snapshot  - 4
    + Received  + 4

    @@ -2,11 +2,11 @@
        "no-debug": Object {
          "create": [Function],
          "meta": Object {
            "docs": Object {
              "category": "Mistake Prevention",
    -         "description": "Don't allow debug() to be committed to the repository. ",
    +         "description": "Don't allow `t.debug()` to be committed to the repository. ",
              "recommended": true,
            },
            "fixable": null,
            "schema": Array [],
          },
    @@ -14,23 +14,23 @@
        "no-only": Object {
          "create": [Function],
          "meta": Object {
            "docs": Object {
              "category": "Mistake Prevention",
    -         "description": "Don't allow test.only to be added to the repository",
    -         "recommended": false,
    +         "description": "Don't allow `test.only` to be added to the repository",
    +         "recommended": true,
            },
            "fixable": null,
            "schema": Array [],
          },
        },
        "no-skip": Object {
          "create": [Function],
          "meta": Object {
            "docs": Object {
              "category": "Mistake Prevention",
    -         "description": "Don't allow test.skip to be added to the repository",
    +         "description": "Don't allow `test.skip` or `fixture.skip` to be added to the repository",
              "recommended": true,
            },
            "fixable": null,
            "schema": Array [],
          },

      2 | 
      3 | it("should have all the rules", () => {
    > 4 |   expect(rules).toMatchSnapshot();
        |                 ^
      5 | });
      6 | 
      7 | it("should have a recommended config with recommended rules", () => {

      at Object.it (tests/index.test.js:4:17)

 › 1 snapshot failed.
 PASS  tests/lib/rules/no-only.js
 PASS  tests/lib/rules/no-skip.js
 PASS  tests/lib/rules/no-debug.js
------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
------------------|---------|----------|---------|---------|-------------------
All files         |     100 |      100 |     100 |     100 |                  
 master           |     100 |      100 |     100 |     100 |                  
  index.js        |     100 |      100 |     100 |     100 |                  
 master/lib/rules |     100 |      100 |     100 |     100 |                  
  no-debug.js     |     100 |      100 |     100 |     100 |                  
  no-only.js      |     100 |      100 |     100 |     100 |                  
  no-skip.js      |     100 |      100 |     100 |     100 |                  
------------------|---------|----------|---------|---------|-------------------

Snapshot Summary
 › 1 snapshot failed from 1 test suite. Inspect your code changes or run `npm test -- -u` to update them.

Test Suites: 1 failed, 3 passed, 4 total
Tests:       1 failed, 16 passed, 17 total
Snapshots:   1 failed, 1 total
Time:        4.596 s
Ran all test suites.

**Comment by benmonro** [comment]
@miherlosev ok updated, please try again.

File: lib/rules/no-debug.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        fixable: null,
+        schema: []
+    },
+
+    create: function (context) {
+        return {
+            "CallExpression[callee.property.name='debug']"(node) {
+                context.report({
+                    node: node.callee.property,
+                    message: 'Do not use debug'
```
Comment (miherlosev) [inline lib/rules/no-debug.js]: I propose to add more context to the error messages:
```js
Do not use the '.debug' action.
Do not use the '.only' hook.
Do not use the '.skip' hook.
```

File: lib/rules/no-only.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create: function (context) {
+
+        return {
+
+            "MemberExpression[object.name=/fixture|test/][property.name='only']"(node) {
```
Comment (miherlosev) [inline lib/rules/no-only.js]: The `.skip` and `.only` hooks can be located not only after the `fixture` or `test` directives.
```js
fixture `Fixture`
	.page('https://example.com')
	.only;

test('test', async () => {});
```

File: lib/rules/no-debug.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            category: "Mistake Prevention",
+            recommended: true
+        },
+        fixable: null,
+        schema: []
+    },
+
+    create: function (context) {
+        return {
+            "CallExpression[callee.property.name='debug']"(node) {
```
Comment (miherlosev) [inline lib/rules/no-debug.js:24]: Can we check that the `.debug` call performed into the `test` function?

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: lib/rules/no-only.js (hunk 2 above)
Comment (benmonro) [reply lib/rules/no-only.js]: I will add more tests for these but I'm pretty sure they would work

File: lib/rules/no-debug.js (hunk 3 above)
Comment (benmonro) [reply lib/rules/no-debug.js:24]: I thought about that but I don't want them in roles or custom functions or page object models either.  It's safer this way and if on the off chance it's reporting incorrectly you can use eslint disable

**Comment by benmonro** [comment]
@miherlosev I addressed your feedback.

**Comment by benmonro** [comment]
@miherlosev ? can we merge this now?

**Comment by benmonro** [comment]
@miherlosev sorry I need these rules and can't wait (we have some tight deadlines).  Decided to move this to a new org called `testcafe-community`

**Pull request closed without merging**